MonoGame.Framework/Vector2.cs

```diff
@@ -756,9 +756,11 @@ public static void Negate(ref Vector2 value, out Vector2 result)
         /// </summary>
         public void Normalize()
         {
-            float val = 1.0f / (float)Math.Sqrt((X * X) + (Y * Y));
-            X *= val;
-            Y *= val;
+			var magnitude = (float)Math.Sqrt((X * X) + (Y * Y));
+			if(magnitude > 1E-05f)
```

Comment:
I know this seems logical to protect against divide by zero...  but this isn't want XNA does.  XNA returns `NaN` if you normalize a zero length vector.  So this change breaks XNA behavior and we cannot merge this as is.

That said... a lot of people make extension methods like `NormalizeSafe` that do that sort of check.

@KonajuGames @dellis1972 - Do you think we should consider adding some sort of "safe" normalization method?  What would be the fallout if it conflicts with people's existing engines that implement extension methods named `NormalizeSafe`?